src/Authentication/JWTAuthenticationHandler.php

```diff
@@ -1,61 +1,50 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace Firesphere\GraphQLJWT\Authentication;
 
+use BadMethodCallException;
+use Exception;
+use Firesphere\GraphQLJWT\Extensions\MemberExtension;
 use Firesphere\GraphQLJWT\Helpers\HeaderExtractor;
+use Firesphere\GraphQLJWT\Helpers\RequiresAuthenticator;
+use OutOfBoundsException;
 use SilverStripe\Control\HTTPRequest;
-use SilverStripe\ORM\ValidationException;
+use SilverStripe\Core\Injector\Injectable;
 use SilverStripe\Security\AuthenticationHandler;
 use SilverStripe\Security\Member;
 use SilverStripe\Security\Security;
 
 /**
  * Class JWTAuthenticationHandler
  *
- *
  * @package Firesphere\GraphQLJWT
  */
 class JWTAuthenticationHandler implements AuthenticationHandler
 {
-
-    /**
-     * @var JWTAuthenticator
-     */
-    protected $authenticator;
-
-    /**
-     * @return mixed
-     */
-    public function getAuthenticator()
-    {
-        return $this->authenticator;
-    }
-
-    /**
-     * @param mixed $authenticator
-     */
-    public function setAuthenticator($authenticator)
-    {
-        $this->authenticator = $authenticator;
-    }
+    use HeaderExtractor;
+    use RequiresAuthenticator;
+    use Injectable;
 
     /**
      * @param HTTPRequest $request
      * @return null|Member
```

Comment:
Return can be removed if return types are in PHP7 style

Reply:
True, unless the return type is one of several types (e.g. `string|int|bool`) or if the return type has a comment. I'd rather just keep it consistently than go through and only have them on some methods.

I might wait until psr-5 is finalised to make a call on this https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md

Reply:
Yep, fair enough :)